cmd/macpassd/dbManager.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"database/sql"
+	"log"
+	"os"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type registrationInstance struct {
+	id     int
+	user   string
+	mac    string
+	start  time.Time
+	end    time.Time
+	isDown bool
+}
+
+func regInstanceToMacRegistration(reg registrationInstance) macRegistration {
+	return macRegistration{mac: reg.mac, reg: registration{reg.user, reg.start, reg.end.Sub(reg.start)}}
+}
+
+func _readRegInstances(row *sql.Rows) []registrationInstance {
+	var macs []registrationInstance
+	var current registrationInstance
+	for row.Next() {
+		row.Scan(&current.id, &current.user, &current.mac, &current.start, &current.end, &current.isDown)
+		macs = append(macs, current)
+	}
+	return macs
+}
+
+func getActive(con *sql.DB) []registrationInstance {
+	selectOutdated := `select * from Log where ? < endTime and isDown = false`
+	row, err := con.Query(selectOutdated, time.Now())
+	if err != nil {
+		log.Fatal(err)
+	}
+	return _readRegInstances(row)
+}
+
+func setOutdated(con *sql.DB) []registrationInstance {
+	selectOutdated := `select * from Log where ? > endTime and isDown = false`
+	row, err := con.Query(selectOutdated, time.Now())
+	if err != nil {
+		log.Fatal(err)
+	}
+	macs := _readRegInstances(row)
+	for i := range macs {
+		id := macs[i].id
+		updateOutdated := `update Log set isDown = true where id = ?`
+		con.Exec(updateOutdated, id)
+	}
+	return macs
+}
```

Comment:
Non un problema della funziona ma più una richiesta: Riusciresti a separare questo metodo in due? Il primo dovrebbe ritornare tutti le `registrationInstance` che devono essere messe down. Il secondo metodo è quello che le prende effettivamente come argomento e le modifica nel database. Una sorta di:
```suggestion
func getOutdated(con *sql.DB) []registrationInstance {
	selectOutdated := `select * from Log where ? > endTime and isDown = false`
	row, err := con.Query(selectOutdated, time.Now())
	if err != nil {
		log.Fatal(err)
	}
	return _readRegInstances(row)
}

func setOutdated(con *sql.DB, macs []registrationInstance) {
	for i := range macs {
		id := macs[i].id
		updateOutdated := `update Log set isDown = true where id = ?`
		con.Exec(updateOutdated, id)
	}
}
```

Non amo troppo il nome del primo metodo, se ti viene qualcosa di meglio fai pure. Questo mi servirebbe perché devo sapere quali `registration` effettivamente devono andare down in modo da riabilitare il firewall su iptables.